### DISCUSSION TITLE
IP Neighbor Discovery Service

### DISCUSSION THREAD

**Discussion by ghost** [dtn7/dtn7-rs#34] [Q&A] [answered]
If I wanted to test the IP Neighbor Discovery Service, how would I go about doing it?

**Comment by gh0st42** [comment, marked as answer]
Just follow the tutorial and check the peers on each node.
Currently, we only have our extended  IPND implementation and completely static peers.
As long as you do not deactive IPND via command line, e.g., by setting discover interval to 0s, it is always used.

Beware, our IPND is using CBOR and not SDNV to be in line with BPv7 and not BPv6 for which the original IPND rfc was written.

**Reply by ghost** [reply]
> Just follow the tutorial and check the peers on each node. Currently, we only have our extended IPND implementation and completely static peers. As long as you do not deactive IPND via command line, e.g., by setting discover interval to 0s, it is always used.
> 
> Beware, our IPND is using CBOR and not SDNV to be in line with BPv7 and not BPv6 for which the original IPND rfc was written.

Is there a way I can check, how the beacons are being sent and received on each node?

**Reply by gh0st42** [reply]
You can activate debug log level on dtnd (`-d`) and look into the log output or you can just look at the "wire" with tcpdump or wireshark. Though, they cannot decode the IPND packets and you must do that yourself.
Other than that, you can check the source code or add whatever debug logging you want there.
